src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -491,4 +491,48 @@ private static void SetAspNetCoreUrls(this IResourceBuilder<ProjectResource> bui
             }
         });
     }
+
+    private static void SetBothPortsEnvVariables(this IResourceBuilder<ProjectResource> builder)
+    {
+        builder.WithEnvironment(context =>
+        {
+            builder.SetOnePortsEnvVariable(context, "HTTP_PORTS", "http");
+            builder.SetOnePortsEnvVariable(context, "HTTPS_PORTS", "https");
+        });
+    }
+
+    private static void SetOnePortsEnvVariable(this IResourceBuilder<ProjectResource> builder, EnvironmentCallbackContext context, string portEnvVariable, string scheme)
+    {
+        if (context.EnvironmentVariables.ContainsKey(portEnvVariable))
+        {
+            // If the user has already set that variable, we don't want to override it.
+            return;
+        }
+
+        var ports = new StringBuilder();
+        var firstPort = true;
+
+        // Turn endpoint ports into a single environment variable.
+        foreach (var annotation in builder.Resource.Annotations.OfType<EndpointAnnotation>())
+        {
+            var port = annotation.TargetPort ?? annotation.Port;
```

Comment:
Is this correct? Or should we only be looking at the TargetPort?

Reply:
Also, should we include 8080 (default container port) on the list for http, in addition to the explicit ports? As in @DamianEdwards's https://github.com/dotnet/aspire-samples/commit/b4d24f87f01d644cefb750dfc431fd02de994ad5

Reply:
Per @davidfowl on Discord: "No we shouldn't include 8080, null endpoint = 8080". So we're good here.